pkg/scaffold/ansible/k8s_status.go

```diff
@@ -0,0 +1,416 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ansible
+
+import (
+	"github.com/operator-framework/operator-sdk/pkg/scaffold/input"
+)
+
+const K8sStatusPythonFile = "library/k8s_status.py"
+
+// K8sStatus - the k8s status module tmpl wrapper
+type K8sStatus struct {
+	input.Input
+}
+
+// GetInput - gets the input
+func (k *K8sStatus) GetInput() (input.Input, error) {
+	if k.Path == "" {
+		k.Path = K8sStatusPythonFile
+	}
+	k.TemplateBody = k8sStatusTmpl
+	return k.Input, nil
+}
+
+func (k K8sStatus) CustomRender() ([]byte, error) {
+	return []byte(k8sStatusTmpl), nil
+}
+
+const k8sStatusTmpl = `#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, print_function
+
+import re
+import copy
+
+from ansible.module_utils.k8s.common import AUTH_ARG_SPEC, COMMON_ARG_SPEC, KubernetesAnsibleModule
+
+try:
+    from openshift.dynamic.exceptions import DynamicApiError
+except ImportError as exc:
+    class KubernetesException(Exception):
+        pass
+
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+
+module: k8s_status
+
+short_description: Update the status for a Kubernetes API resource
+
+version_added: "2.7"
+
+author: "Fabian von Feilitzsch (@fabianvf)"
+
+description:
+  - Sets the status field on a Kubernetes API resource. Only should be used if you are using Ansible to
+    implement a controller for the resource being modified.
+
+options:
+  status:
+    type: dict
+    description:
+    - A object containing ` + "`key: value`" + ` pairs that will be set on the status object of the specified resource.
```

Comment:
I know how weird this looks.... but the python file has backticks in it. This was the cleanest way I could get the entire file to be a single string.